### PR TITLE
Features/admin zone list

### PR DESCRIPTION
**Pull request by AlessioVerardo**
Hi, 
This PR has the following features: 
- The events are accessible by admin (and grouped by zones) 
- Refactoring last database objects
- Not any listener in the database, all match the new structure made by @Rygone with generic methods
- Add tests for ItemDatabase, ZoneManagementListActivity

Cheers

### CODE REVIEW

File: app/src/main/res/layout/card_event_by_zone.xml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/tvEventTitle" />
+
+        <TextView
+            android:id="@+id/tvEventEndDate"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="8dp"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/tvEventStartDate" />
```
Comment (Feez2403) [inline app/src/main/res/layout/card_event_by_zone.xml:67]: I think it looks a bit better like this ;)

```suggestion
        <TextView
            android:id="@+id/tvEventTitle"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:layout_marginStart="16dp"
            android:layout_marginTop="10dp"
            android:textAppearance="@style/TextAppearance.AppCompat.Large"
            app:layout_constraintStart_toStartOf="parent"
            app:layout_constraintTop_toTopOf="parent" />

        <ImageButton
            android:id="@+id/idEditEventButton"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            app:layout_constraintEnd_toEndOf="parent"
            app:layout_constraintTop_toTopOf="parent"
            app:srcCompat="@android:drawable/ic_menu_edit"
            tools:ignore="ContentDescription" />

        <ImageButton
            android:id="@+id/idDeleteEventButton"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            app:layout_constraintBottom_toBottomOf="parent"
            app:layout_constraintEnd_toEndOf="parent"
            app:layout_constraintTop_toBottomOf="@+id/idEditEventButton"
            app:layout_constraintVertical_bias="1.0"
            app:srcCompat="@android:drawable/ic_delete"
            tools:ignore="ContentDescription" />

        <TextView
            android:id="@+id/tvEventStartDate"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:layout_marginStart="16dp"
            android:layout_marginTop="8dp"
            app:layout_constraintStart_toStartOf="parent"
            app:layout_constraintTop_toBottomOf="@+id/tvEventTitle" />

        <TextView
            android:id="@+id/tvEventEndDate"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:layout_marginStart="16dp"
            android:layout_marginTop="8dp"
            app:layout_constraintStart_toStartOf="parent"
            app:layout_constraintTop_toBottomOf="@+id/tvEventStartDate" />
```

File: app/src/main/res/layout/activity_event_management_list.xml [hunk 2]
```diff
@@ -20,13 +20,23 @@
...
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/recycler_events_list_admin"
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/btnNewEvent"
+        app:layout_constraintVertical_bias="0.028"
```
Comment (Feez2403) [inline app/src/main/res/layout/activity_event_management_list.xml:40]: you can remove this line 
```suggestion
        
```

File: app/src/main/java/com/github/sdpteam15/polyevents/view/adapter/EventListAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+            }
+            notifyDataSetChanged()
+        }
+    }
+
+
+    abstract inner class CustomViewHolder<T>(view: View) :
+            RecyclerView.ViewHolder(view) {
+        abstract fun bind(value: T)
+        abstract fun unbind()
```
Comment (Feez2403) [inline app/src/main/java/com/github/sdpteam15/polyevents/view/adapter/EventListAdapter.kt:51]: You don't use the unbind method in both subclasses, you can remove it here

File: app/src/main/java/com/github/sdpteam15/polyevents/view/adapter/EventListAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+            }
+            //Should not happen
+            else -> throw IllegalArgumentException("wrong view type $viewType")
+        }
+    }
+
+    override fun onViewRecycled(holder: CustomViewHolder<*>) {
+        super.onViewRecycled(holder)
+        holder.unbind()
+    }
```
Comment (Feez2403) [inline app/src/main/java/com/github/sdpteam15/polyevents/view/adapter/EventListAdapter.kt:132]: if you delete the unbind, you can remove these lines too
```suggestion
    
```

File: app/src/main/java/com/github/sdpteam15/polyevents/view/adapter/EventListAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+import com.github.sdpteam15.polyevents.model.observable.ObservableMap
+
+/**
+ * Recycler Adapter for the list of items
+ * Shows each item with its available quantity
+ * @param context context of parent view used to inflate new views
+ * @param lifecycleOwner parent to enable observables to stop observing when the lifecycle is closed
+ * @param availableItems Map of each available item grouped by types
+ * @param mapSelectedItems Item counts for each item
+ */
```
Comment (Feez2403) [inline app/src/main/java/com/github/sdpteam15/polyevents/view/adapter/EventListAdapter.kt:24]: Check all the comments of this class x)

File: app/src/main/java/com/github/sdpteam15/polyevents/view/activity/admin/EventManagementListActivity.kt [hunk 6]
```diff
@@ -3,23 +3,94 @@ package com.github.sdpteam15.polyevents.view.activity.admin
...
+    private fun setupListeners() {
+        modifyListener = { s: String ->
+            val intent = Intent(this, EventManagementActivity::class.java)
+            intent.putExtra(EVENT_ID_INTENT, s)
+            startActivity(intent)
+        }
+
+        deleteListener = { zoneId: String, event: Event ->
+            val builder = AlertDialog.Builder(this)
+            builder.setMessage("Are you sure to want to delete " + event.eventName + " (this action is not reversible)")
```
Comment (Feez2403) [inline app/src/main/java/com/github/sdpteam15/polyevents/view/activity/admin/EventManagementListActivity.kt:73]: Nice ! I like it, we should use for every permanent deletion

**Review by Feez2403 — APPROVED** [review]
Great implementation and nice refactor, I can't wait to merge my branch with master ;-;

File: app/src/main/res/layout/card_event_by_zone.xml (hunk 1 above)
Comment (AlessioVerardo) [reply app/src/main/res/layout/card_event_by_zone.xml]: Yes, you're correct !

File: app/src/main/java/com/github/sdpteam15/polyevents/view/adapter/EventListAdapter.kt (hunk 5 above)
Comment (AlessioVerardo) [reply app/src/main/java/com/github/sdpteam15/polyevents/view/adapter/EventListAdapter.kt]: Oupsi x)

File: app/src/main/res/layout/activity_event_management_list.xml (hunk 2 above)
Comment (AlessioVerardo) [reply app/src/main/res/layout/activity_event_management_list.xml]: Yep

File: app/src/main/java/com/github/sdpteam15/polyevents/view/adapter/EventListAdapter.kt (hunk 4 above)
Comment (AlessioVerardo) [reply app/src/main/java/com/github/sdpteam15/polyevents/view/adapter/EventListAdapter.kt]: Okay 👍

File: app/src/main/java/com/github/sdpteam15/polyevents/view/activity/admin/EventManagementListActivity.kt (hunk 6 above)
Comment (AlessioVerardo) [reply app/src/main/java/com/github/sdpteam15/polyevents/view/activity/admin/EventManagementListActivity.kt:73]: Yep, in the next sprint, we can extract the content of the listener and add it in the Helper Functions.

**Review by Rygone — APPROVED** [review]
goode refactor of the db

**Pull request merged**